### PR TITLE
Watch Multiple Namespaces

### LINKED ISSUE #228: Deploy druid cluster across multiple namespace
Thanks for the awesome operator. I had a question, and wondering if you could help.

I want to deploy the Druid operator in one namespace (druid-dev) and the Druid cluster in another namespace(druid-stag). When I hardcode the target namespace name (druid-stag) in the operator yaml in the watch namespace section and create a role in the stag cluster, it works. 

But I need to add multiple namespaces to it. 

When I try to leave the watch namespace section empty (value: "") to support any namespace, it throws an error:
```E0713 20:08:43.335828       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/reflector.go:125: Failed to list *v1alpha1.Druid: druids.dataplatform.xxx.com is forbidden: User "system:serviceaccount:druid-dev:druid-operator" cannot list resource "druids" in API group "xxx" at the cluster scope
E0713 20:09:19.536133       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.18.2/tools/cache/reflector.go:125: Failed to list *v1alpha1.Druid: druids.dataplatform.xxx.com is forbidden: User "system:serviceaccount:druid-dev:druid-operator" cannot list resource "druids" in API group "dataplatform.xxx.com" at the cluster scope
```
Our company has a Kubernetes offering that does not allow the usage of clusterRoles. Any idea how we can do it with Roles and Role Binding. The suggestion is to use role and rolebinding in the target namespace, but I am not sure how to approach it.

Or can we pass multiple namespaces in the watch namespace section? If this works, then it would be fine too

### PR DESCRIPTION
**Pull request by AdheipSingh**
Fixes #228 .

### Description
- Deploy druid cluster in multiple namespaces.

<hr>

This PR has:
- [x] been tested on a real K8S cluster to ensure creation of a brand new Druid cluster works.
- [x] been tested for backward compatibility on a real K*S cluster by applying the changes introduced here on an existing Druid cluster. If there are any backward incompatible changes then they have been noted in the PR description.
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added documentation for new or modified features or behaviors.

<hr>

##### Key changed/added files in this PR
 * `main.go`

@himanshug @nishantmonu51

### CODE REVIEW

**Comment by AdheipSingh** [comment]
@cintoSunny this should make sure druid operator can watch multiple namespaces, WATCH_NAMESPACE will be an env as it now, but will support slice of strings

File: main.go [hunk 1]
```diff
@@ -63,3 +66,17 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func watchNamespaceCache() cache.NewCacheFunc {
```
[the remark this answers is not in the document]
Comment (AdheipSingh) [reply main.go:70]: i had tried to write this function into the smoke test suite, but cache needs a real k8s cluster, and will exit 1. 
what i can do is extract, the string manipulation code into a util function and write a simple unit test for it.

Thanks

**Review by AdheipSingh — COMMENTED** [review]
add multi-ns support in helm charts

**Comment by churromorales** [comment]
is there a reason why this has not been merged?  This is a very useful PR.

File: main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (AdheipSingh) [reply main.go:87]: will take it in e2e PR

**Pull request merged**